logic/src/commonTest/kotlin/com/wire/kalium/logic/feature/featureConfig/GetFileSharingStatusUseCaseTest.kt

```diff
@@ -0,0 +1,123 @@
+        assertEquals(arrangement.fileSharingModel, actual.fileSharingModel)
+
+        verify(arrangement.featureConfigRepository)
+            .suspendFunction(arrangement.featureConfigRepository::getFileSharingFeatureConfig)
+            .wasInvoked(exactly = once)
+    }
+
+    @Test
+    fun givenRepositoryCallFailWithInvalidCredentials_thenOperationDeniedIsReturned() = runTest {
+        // Given
+        val operationDeniedException = TestNetworkException.operationDenied
+        val (arrangement, getFileSharingStatusUseCase) = Arrangement()
+            .withGetFileSharingStatusErrorResponse(operationDeniedException)
+            .arrange()
+
+        // When
+        val actual = getFileSharingStatusUseCase.invoke()
+
+        // Then
+        assertTrue(actual is GetFileSharingStatusResult.Failure.OperationDenied)
+        assertEquals(arrangement.operationDeniedFailure, actual)
+
+        verify(arrangement.featureConfigRepository)
+            .suspendFunction(arrangement.featureConfigRepository::getFileSharingFeatureConfig)
+            .wasInvoked(exactly = once)
+    }
+
+    @Test
+    fun givenRepositoryCallFailWithUserThatNotInTheTeam_thenNoTeamIsReturned() = runTest {
+        // Given
+        val noTeamException = TestNetworkException.noTeam
+        val (arrangement, getFileSharingStatusUseCase) = Arrangement()
+            .withGetFileSharingStatusErrorResponse(noTeamException)
+            .arrange()
+
+        // When
+        val actual = getFileSharingStatusUseCase.invoke()
+
+        // Then
+        assertTrue(actual is GetFileSharingStatusResult.Failure.NoTeam)
+        assertEquals(arrangement.noTeamFailure, actual)
+
+        verify(arrangement.featureConfigRepository)
+            .suspendFunction(arrangement.featureConfigRepository::getFileSharingFeatureConfig)
+            .wasInvoked(exactly = once)
+    }
+
+
+    private class Arrangement {
+        val fileSharingModel = FileSharingModel("locked", "enabled")
+        val operationDeniedFailure = GetFileSharingStatusResult.Failure.OperationDenied
+        val noTeamFailure = GetFileSharingStatusResult.Failure.NoTeam
+
+        @Mock
+        val featureConfigRepository = mock(classOf<FeatureConfigRepository>())
+
+        @Mock
+        val userConfigRepository = mock(classOf<UserConfigRepository>())
+
+
+        val getFileSharingStatusUseCase =
+            GetFileSharingStatusUseCaseImpl(userConfigRepository, featureConfigRepository)
+
+        fun withSuccessfulResponse(): Arrangement {
+            given(userConfigRepository)
+                .function(userConfigRepository::setFileSharingStatus)
+                .whenInvokedWith(any())
+                .thenReturn(Either.Right(Unit))
+
+            given(featureConfigRepository)
+                .suspendFunction(featureConfigRepository::getFileSharingFeatureConfig).whenInvoked()
+                .thenReturn(Either.Right(fileSharingModel))
+            return this
+        }
```

Comment:
Maybe we could provide the `fileSharingModel` as an expected parameter from the arrangement setup?
```suggestion
        fun withSuccessfulResponse(val expectedFileSharingModel): Arrangement {
            given(userConfigRepository)
                .function(userConfigRepository::setFileSharingStatus)
                .whenInvokedWith(any())
                .thenReturn(Either.Right(Unit))

            given(featureConfigRepository)
                .suspendFunction(featureConfigRepository::getFileSharingFeatureConfig).whenInvoked()
                .thenReturn(Either.Right(expectedFileSharingModel))
            return this
        }
```
That way we can check that the actual response object matches our initial expectations and that the arrangement does what it is meant to, instead of "trusting" the arrangement internal object mapping.